contracts/Example2.sol

```diff
@@ -1,11 +1,25 @@
+//SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+/// @title A contract to show gas optimization by using temp variables and calldata
+/// @author Ashwin Rachakonda 💯💯😎💯💯
+/// @notice A contract to show gas optimization by using temp variables and calldata
+
+//Function call gas estimates
+//50355 - original
+//48534 - after calldata
+//48499 - after temp variable to keep array length
+//47719 - after using temp variable to calculate sum
+
 contract Example2 {
     uint public counter;
- 
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+    
+    function incrementBy(uint[] calldata arr) public {
+        uint arrLength =  arr.length;
+        uint tempSum = counter;
+        for (uint idx = 0; idx < arrLength; idx++){
+            tempSum += arr[idx];
         }
+        counter += tempSum;
```

Comment:
Shouldn't this just be `counter = tempSum` since you have already added the value of counter when declaring `uint tempSum = counter;`

Reply:
thanks @shubham-kanodia for the review.